server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1397,6 +1377,63 @@ public boolean performConsumerUpdates(ConsumerDTO updated, Consumer toUpdate,
         return changesMade;
     }
 
+    private boolean updateEnvironment(ConsumerDTO updated, Consumer toUpdate) {
+        boolean changesMade = false;
+        boolean environmentFound = false;
+        Environment e = null;
+        String environmentId = updated.getEnvironment() == null ? null : updated.getEnvironment().getId();
+        String environmentName = updated.getEnvironment() == null ? null : updated.getEnvironment().getName();
+
+        if (environmentName != null && environmentId == null) {
+            List<Environment> environment = environmentCurator.listForOwnerByName(toUpdate.getOwner(),
+                environmentName).list();
+
+            if (!environment.isEmpty() && environment.size() == 1) {
+                e = environment.get(0);
+                environmentId = environment.get(0).getId();
+                environmentFound = true;
+            }
+            else {
+                throw new NotFoundException(i18n.tr(
+                        "Environment with name \"{0}\" could not be found.", environmentName));
+            }
+
+        }
+
+        if (environmentId != null && (toUpdate.getEnvironmentId() == null ||
+            !toUpdate.getEnvironmentId().equals(environmentId))) {
+
+            if (!environmentFound) {
```

Comment:
environmentFound will never be true at this point, so this check doesn't do anything. If the intent is only to do this lookup if the environment has not yet been looked up (i.e. we already did that lookup by name), then this whole section needs a refactor around that idea.